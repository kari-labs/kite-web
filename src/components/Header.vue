<template>
    <div>
        <nav-drawer ref="navDrawer"></nav-drawer>
        <v-toolbar app relative dark clipped-left color="primary">
            <v-toolbar-side-icon @click.stop="toggleDrawer" temporary class="secondary-text" aria-label="Toggle side navigation drawer"></v-toolbar-side-icon>
            <v-toolbar-title class="secondary-text" @click="pushDash">NEIT Kite</v-toolbar-title>
            <v-spacer></v-spacer>
            <v-toolbar-items class="hidden-sm-and-down">
                <v-btn flat color="secondary" slot="activator" :to="''" aria-label="Container Dashboard">Container Dashboard</v-btn>
                <v-btn flat color="secondary" aria-label="Report a Bug">Report Bug</v-btn>
            </v-toolbar-items>
        </v-toolbar>
    </div>
</template>

<script>
    export default {
        methods: {
            toggleDrawer() {
                //Toggles the navDrawer open/closed only if the navDrawer component
                //has finished downloading.
                this.$refs.navDrawer.drawer = (typeof this.$refs.navDrawer.drawer == 'undefined') ? false : !this.$refs.navDrawer.drawer
            },
            pushDash() {
                this.$router.push({ name: 'listContainers'})
            }
        },
        components: {
            'nav-drawer': () => import(/* webpackChunkName: "navDrawer" */ '@/components/NavDrawer.vue')
        }
    }
</script>

<style lang="stylus" scoped>
//Fixes bug where Vuetify secondary changes color of background but not text
.v-toolbar__side-icon.secondary-text
    color #FCB741 !important

//Disables ability to select toolbar title text
.v-toolbar__title
    user-select none
</style>
