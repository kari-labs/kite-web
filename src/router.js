import Vue from 'vue'
import Router from 'vue-router'

Vue.use(Router)

export default new Router({
  mode: 'history',
  base: '/',
  routes: [
    {
        path: '/',
        name: 'listContainers',
        component: () => import(/* webpackChunkName: "containerDashboard" */ './views/ContainerDashboard.vue')
    }
  ]
})
