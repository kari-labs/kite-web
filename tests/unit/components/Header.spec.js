import { mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import router from '@/router'
import Header from '@/components/Header.vue'

describe('Header', () => {
    let wrapper

    beforeEach(() => {
        Vue.use(router)
        Vue.use(Vuetify)

        wrapper = mount(Header, { router })
    })

    it('should render Vuetify components', () => {
        expect(wrapper.contains('nav.v-toolbar')).toBe(true)
    })

    it('should render NavDrawer', () => {
        expect(wrapper.contains('aside.v-navigation-drawer')).toBe(true)
    })

    it('should open a closed NavDrawer on click', () => {
        wrapper.find('button.v-toolbar__side-icon').trigger('click')
        expect(wrapper.vm.$refs.navDrawer.drawer).toBe(true)
    })

    it('should NOT open a NavDrawer while loading', () => {
        wrapper.vm.$refs.navDrawer.drawer = undefined

        wrapper.vm.toggleDrawer()
        expect(wrapper.vm.$refs.navDrawer.drawer).toBe(false)
    })

    it('should change route on toolbar title click', () => {
        wrapper.find('div.v-toolbar__title').trigger('click')
        expect(wrapper.vm.$route.path).toBe('/' || '')
    })

    it('calling toggleDrawer will change value of navDrawer.drawer', () => {
        const initialValue = wrapper.vm.$refs.navDrawer.drawer

        wrapper.vm.toggleDrawer()
        expect(wrapper.vm.$refs.navDrawer.drawer).toBe(!initialValue)
    })

    it('renders all elements correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
})