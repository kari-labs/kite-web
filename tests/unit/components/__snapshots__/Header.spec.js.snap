// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders all elements correctly 1`] = `
<div>
  <aside
    class="accent v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light"
    style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); transform: translateX(-300px); width: 300px;"
  >
    <div
      class="v-list pt-0 v-list--dense theme--light"
    >
      <div>
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__action"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              dashboard
            </i>
          </div>
           
          <div
            class="v-list__tile__content"
          >
            <div
              class="v-list__tile__title"
            >
              Home
            </div>
          </div>
        </a>
      </div>
      <div>
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__action"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              bug_report
            </i>
          </div>
           
          <div
            class="v-list__tile__content"
          >
            <div
              class="v-list__tile__title"
            >
              Report Bug
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
   
  <nav
    class="v-toolbar v-toolbar--clipped v-toolbar--fixed theme--dark primary"
    relative=""
    style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        aria-label="Toggle side navigation drawer"
        class="secondary-text v-toolbar__side-icon v-btn v-btn--icon theme--dark"
        temporary=""
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--dark"
          >
            menu
          </i>
        </div>
      </button>
       
      <div
        class="v-toolbar__title secondary-text"
      >
        NEIT Kite
      </div>
       
      <div
        class="spacer"
      />
       
      <div
        class="v-toolbar__items hidden-sm-and-down"
      >
        <button
          aria-label="Container Dashboard"
          class="v-btn v-btn--flat theme--dark secondary--text"
          slot="activator"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            Container Dashboard
          </div>
        </button>
         
        <button
          aria-label="Report a Bug"
          class="v-btn v-btn--flat theme--dark secondary--text"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            Report Bug
          </div>
        </button>
      </div>
    </div>
  </nav>
</div>
`;
